src/bcf/record.rs

```diff
@@ -335,69 +335,168 @@ impl Record {
-                return true;
-            }
+    /// # Example
+    /// ```rust
+    /// # use rust_htslib::bcf::{Format, Header, Writer};
+    /// # use tempfile::NamedTempFile;
+    /// # let tmp = tempfile::NamedTempFile::new().unwrap();
+    /// # let path = tmp.path();
+    /// let mut header = Header::new();
+    /// header.push_record(br#"##FILTER=<ID=foo,Description="sample is a foo fighter">"#);
+    /// # let vcf = Writer::from_path(path, &header, true, Format::VCF).unwrap();
+    /// # let mut record = vcf.empty_record();
+    /// assert!(record.has_filter(b"PASS"));
+    /// assert!(record.has_filter(b"."));
+    ///
+    /// record.push_filter(b"foo").unwrap();
+    /// assert!(record.has_filter(b"foo"));
+    /// assert!(!record.has_filter(b"PASS"))
+    /// ```
+    pub fn has_filter(&self, flt_id: &[u8]) -> bool {
+        let filter_c_str = ffi::CString::new(flt_id).unwrap();
+        match unsafe {
+            htslib::bcf_has_filter(
+                self.header().inner,
+                self.inner,
+                filter_c_str.as_ptr() as *mut i8,
+            )
+        } {
+            1 => true,
+            _ => false,
         }
-        false
     }
 
-    /// Set the given filters IDs to the FILTER column.
+    /// Set the given filter IDs to the FILTER column.
     ///
-    /// Setting an empty slice removes all filters.
+    /// Setting an empty slice removes all filters and sets `PASS`.
     ///
-    /// # Arguments
+    /// # Example
+    /// ```rust
+    /// # use rust_htslib::bcf::{Format, Header, Writer};
+    /// # use tempfile::NamedTempFile;
+    /// # let tmp = tempfile::NamedTempFile::new().unwrap();
+    /// # let path = tmp.path();
+    /// let mut header = Header::new();
+    /// header.push_record(br#"##FILTER=<ID=foo,Description="sample is a foo fighter">"#);
+    /// header.push_record(br#"##FILTER=<ID=bar,Description="a horse walks into...">"#);
+    /// # let vcf = Writer::from_path(path, &header, true, Format::VCF).unwrap();
+    /// # let mut record = vcf.empty_record();
+    /// assert!(record.has_filter(b"PASS"));
+    /// record.set_filters(&[b"foo", b"bar"]).unwrap();
+    /// assert!(record.has_filter(b"foo"));
+    /// assert!(record.has_filter(b"bar"));
+    /// assert!(!record.has_filter(b"PASS"));
+    /// record.set_filters(&[]).unwrap();
+    /// assert!(record.has_filter(b"PASS"));
+    /// assert!(!record.has_filter(b"foo"));
+    /// // 'baz' isn't in the header
+    /// assert!(record.set_filters(&[b"foo", b"baz"]).is_err())
+    /// ```
+    ///
+    /// # Errors
+    /// If any of the filter IDs do not exist in the header, an [`Error::BcfUnknownID`] is returned.
     ///
-    /// - `flt_ids` - The identifiers of the filter values to set.
-    pub fn set_filters(&mut self, flt_ids: &[Id]) {
-        let mut flt_ids: Vec<i32> = flt_ids.iter().map(|x| **x as i32).collect();
+    pub fn set_filters(&mut self, flt_ids: &[&[u8]]) -> Result<()> {
```

Comment:
Maybe rename `flt_ids` to something like `flt_names` (as in `name_to_id()`)? Although ID refers to the respective field where the filter name is given in the header, it also refers to the number assigned to that filter name internally. So maybe `flt_names` is less ambiguous?